### PR TITLE
Author page fix

### LINKED ISSUE #125: FRONTEND BUG: Follow Status Visual Bug
If I am a friends with another author, when I go to their author profile I expect to see an unfollow button. If I am not friends with another author, and I click to go to their profile, the unfollow button stays instead of the follow button.

1) Follow an author on the app.
2) Go to their profile and view their friends
3) If there is an author you are not following, click their profile to go to their page
4) The unfollow button should show up instead of a follow button

image

### LINKED ISSUE #126: Refresh author page bug
refreshing author page kills the page because of this.props.location.state does not exist

### LINKED ISSUE #128: follow button bug
in the author page, if they are friends, it shows the follow button

### PR DESCRIPTION
**Pull request by hiufungk**
resolves #125 
resolves #126 
resolves #128 

Fixed author page to be dependent on author id in url
Fixed follow button bug

backend url settings has been changed from http to https
THIS WILL BREAK YOUR EXISTING AUTHORS, FOLLOWERS since the endpoint compares the full author ID.
Everyone is recommended to remove their existing authors, followers just to be safe
or just delete dev db if you are lazy (thats what I did)

### CODE REVIEW

**Comment by derrick-w** [comment]
bless

**Review by derrick-w — APPROVED** [review]
looks good, i leave it to our quality assurance engineer to try to break it

File: frontend/src/pages/Author.jsx [hunk 1]
```diff
@@ -82,20 +90,20 @@ class Author extends Component {
...
         }
 
-        else if (this.state.isFollowing && !this.state.isSelf) {
-            followbutton = <Button negative onClick={this.sendUnfollowRequest}><Icon name = "user times"/>Unfollow</Button>
+        if (this.state.isFollowing || this.state.isFriends) {
+            button = <Button negative onClick={this.sendUnfollowRequest}><Icon name = "user times"/>Unfollow</Button>
         }
 
-        return <div>{followbutton}</div>;
+        return <div>{button}</div>;
```
Comment (shchau) [inline frontend/src/pages/Author.jsx:106]: Naming could be better, but I can't think of a better name uwu

File: frontend/src/components/FriendsListComponent.jsx [hunk 2]
```diff
@@ -30,11 +30,7 @@ class FriendListComponent extends Component {
 			return(
 			<div>
 				<i className="server icon"></i>
-				<Link to={
-							{pathname: "/author/"+utils.getStripedEscapedAuthorId(authorObj.url.substring(7,)),
-							  state: {
-							  	fullAuthorId: authorObj.url,
-							  }}}>
+				<Link to={"/author/"+utils.getStripedEscapedAuthorId(authorObj.url.substring(7,))}>
```
Comment (shchau) [inline frontend/src/components/FriendsListComponent.jsx:33]: Zebras have stripes. Our clothes, much like our strings, get stripped. 

```suggestion
				<Link to={"/author/"+utils.getStrippedEscapedAuthorId(authorObj.url.substring(7,))}>
```

File: frontend/src/util/utils.js [hunk 3]
```diff
@@ -10,6 +10,15 @@ export default class utils {
         throw new Error("getStripedEscapedAuthorId invalid argument");
     }
 
+    static unEscapeAuthorId(escapedAuthorId) {
+        if(escapedAuthorId) {
+            let authorId = decodeURIComponent(escapedAuthorId);
+            authorId = "https://" + authorId;
+            return authorId;
+        }
+        throw new Error("unEscapeAuthorId invalid argument");
```
Comment (forgeno) [inline frontend/src/util/utils.js:19]: image
 Looks good doe

**Review by forgeno — APPROVED** [review]

File: frontend/src/pages/Author.jsx (hunk 1 above)
Comment (derrick-w) [reply frontend/src/pages/Author.jsx:106]: make followbutton great again

**Comment by htruong4** [comment]
Just tested manually, merging it in now.

**Pull request merged**